### ISSUE TITLE
SAE streaming Arrow-Schur: undamped evidence/log-det factorization of a rank-1 (p_out=1) / indefinite per-row H_tt at base ridge_t=0 exceeds factor_one_row's escalation cap

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#847]
## Summary

Two SAE streaming Arrow-Schur tests fail with a per-row latent Hessian that is not positive-definite when factored at **base ridge 0**:

- `src/terms/sae_manifold.rs::tests::streaming_exact_reml_matches_full_batch_reml_small_sae` (`sae_manifold.rs:9448`)
  - `Err: "arrow-Schur: per-row H_tt^(1) Cholesky failed: ... row 1 H_tt ... non-PD at base ridge_t=0"`
- `src/terms/sae_manifold.rs::tests::streaming_plan_routes_by_memory_budget_with_identical_logdet` (`sae_manifold.rs:9532`)
  - `ArrowSchurError::PerRowFactorFailed { row: 0, ... }` — "H_tt is non-PD at base ridge 0e0"

Reproduced on committed main in Rust CI run 27084917256.

## Root cause (file:line): undamped evidence/log-det factorizations factor per-row `H_tt` at `ridge_t = 0`, and `factor_one_row`'s escalation cap cannot lift a strongly-indefinite / rank-deficient block

Both tests use `small_two_atom_periodic_term()` (`sae_manifold.rs:9393`): a **Softmax K=2** assignment over two 1-D periodic atoms, `p_out = 1`, `n = 5`.

The per-row latent block `H_tt^(i)` is assembled as a Gauss-Newton data term `J^T J` (`sae_manifold.rs:4077`), plus the assignment-prior Hessian diagonal (`sae_manifold.rs:4094`/`4099`), plus the PSD-majorized ARD curvature. With `p_out = 1` the data Gram `J^T J` is **rank-1** over the `q = assignment_dim + total_coord_dim`-dimensional row block, so `H_tt` is rank-deficient (PSD-singular) before the prior, and the softmax assignment-sparsity `hessian_diag` (`assignment_prior_grad_hdiag`, `sae_manifold.rs:8458`) can add a negative diagonal on the logit coordinate, making the block genuinely indefinite at points off the inner optimum.

The two paths factor this block **undamped at `ridge_t = 0`**:

- `streaming_plan_routes` calls `solve_arrow_newton_step_with_options(&sys, 0.0, 0.0, &options)` directly on `term0` **without first running the inner joint fit** (`sae_manifold.rs:9565`), so `H_tt` is evaluated at the *initial* (non-stationary) iterate where it is indefinite.
- `streaming_exact_reml` runs the inner fit first, then `streaming_exact_arrow_log_det` factors each chunk with `reduced_schur_and_log_det_tt(0.0, 0.0, ...)` (`sae_manifold.rs:4893`).

Both reach `factor_one_row` (`arrow_schur.rs:1458`) with `ridge_t = 0`. Its per-row escalation cap is

```rust
let diag_scale = row_block_diag_scale(row, d);          // = max(|diag|, 1) ≥ 1   (arrow_schur.rs:1371)
let ridge_cap = ridge_t.max(1.0e-12 * diag_scale) * 1.0e12;   // = ~1.0 when ridge_t == 0   (arrow_schur.rs:1502)
```

so with `ridge_t = 0` the escalation can only lift the ridge to ~`diag_scale` (≈ O(1) here). A block whose most-negative pivot exceeds that cap surfaces `PerRowFactorFailed` (`arrow_schur.rs:1560`) instead of being regularised.

## Why the dense REML baseline passes but these fail

`reml_criterion_with_cache` (the `full` baseline in `streaming_exact_reml`) also factors undamped at ridge 0 (`sae_manifold.rs:4667`), but only **after** `run_joint_fit_arrow_schur` converges the inner `(t, β)` state, where `H_tt` is PD. The fit path itself tolerates indefinite rows because it routes through `solve_with_lm_escalation` (`sae_manifold.rs:6657`, see the #163 regression test at `sae_manifold.rs:9599`), which damps both ridges. The undamped evidence/log-det callers have no such escalation, so:

- the **direct-assembly** path (`streaming_plan_routes`) fails because it never converges the inner state, and
- the **streaming-exact** path fails because its converged row-1 block is barely-/non-PD and the `ridge_t = 0` escalation cap is too low.

## Discrimination (genuine code behavior vs test setup)

This is a genuine gap in the undamped log-det path: it cannot regularise a PSD-singular (rank-deficient, `p_out = 1`) or mildly-indefinite per-row block when the caller passes `ridge_t = 0`, whereas the fit path can. Either the evidence/log-det path should pin a nonzero floor ridge derived from the block's diagonal scale (so the rank-deficient `J^TJ` block factors), or `streaming_plan_routes` should converge the inner state (or pass a nonzero `ridge_t`) before the direct factorization, matching how `reml_criterion_with_cache` reaches a PD block. The streaming and dense logdets are meant to be bit-identical (`assert_abs_diff_eq! epsilon = 1e-8`), so both must use the same ridge regime.
